controllers/resources.go

```diff
@@ -0,0 +1,283 @@
+										SecretKeyRef: &corev1.SecretKeySelector{
+											Key: "password",
+											LocalObjectReference: corev1.LocalObjectReference{
+												Name: fmt.Sprintf("%s-%s", localDeploymentResourceName, dbConnectionSecretRef.Name),
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}, scheme)
+	if err != nil {
+		return nil, err
+	}
+	deploymentAppResourceInLocal := workload.KubernetesApplicationResourceTemplate{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      localDeploymentResourceName,
+			Namespace: wp.GetNamespace(),
+			Labels:    GetLocalResourceSelector(wp),
+		},
+		Spec: workload.KubernetesApplicationResourceSpec{
+			Template: deploymentInRemote,
+			Secrets: []corev1.LocalObjectReference{
+				{Name: dbConnectionSecretRef.Name},
+			},
+		},
+	}
+
+	serviceInRemote, err := ConvertToUnstructured(&corev1.Service{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "Service",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      wp.GetName(),
+			Namespace: namespaceInRemote.GetName(),
+			Labels:    labelsInRemote,
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				{
+					Port: int32(80),
+				},
+			},
+			Selector: labelsInRemote,
+			Type:     corev1.ServiceTypeLoadBalancer,
+		},
+	}, scheme)
+	if err != nil {
+		return nil, err
+	}
+	serviceAppResourceInLocal := workload.KubernetesApplicationResourceTemplate{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-service-%s-%s", resourcePrefix, wp.GetNamespace(), wp.GetName()),
+			Namespace: wp.GetNamespace(),
+			Labels:    GetLocalResourceSelector(wp),
+		},
+		Spec: workload.KubernetesApplicationResourceSpec{
+			Template: serviceInRemote,
+		},
+	}
+	return &workload.KubernetesApplication{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            wp.GetName(),
+			Namespace:       wp.GetNamespace(),
+			Labels:          GetLocalResourceSelector(wp),
+			OwnerReferences: []metav1.OwnerReference{GetOwnerReference(wp)},
+		},
+		Spec: workload.KubernetesApplicationSpec{
+			ResourceSelector: &metav1.LabelSelector{
+				MatchLabels: GetLocalResourceSelector(wp),
+			},
+			ClusterSelector: &metav1.LabelSelector{
+				MatchLabels: GetLocalResourceSelector(wp),
+			},
+			ResourceTemplates: []workload.KubernetesApplicationResourceTemplate{
+				namespaceAppResourceInLocal,
+				deploymentAppResourceInLocal,
+				serviceAppResourceInLocal,
+			},
+		},
+	}, nil
+}
+
+func GetLocalResourceSelector(wp v1alpha1.WordpressInstance) map[string]string {
+	return map[string]string{
+		localResourceSelectorKeyName: wp.GetName(),
+	}
+}
+
+func GetOwnerReference(wp v1alpha1.WordpressInstance) metav1.OwnerReference {
+	return metav1.OwnerReference{
+		APIVersion: wp.APIVersion,
+		Kind:       wp.Kind,
+		Name:       wp.GetName(),
+		UID:        wp.GetUID(),
+	}
+}
+
+func ConvertToUnstructured(o metav1.Object, scheme *runtime.Scheme) (*unstructured.Unstructured, error) {
```

Comment:
@suskin actually I'm not sure whether that comment would be useful. It's a classic type conversion like json marshal that's needed for the corresponding field's type.

Reply:
@muvaf it's okay if it can be used for a different reason; that comment is still useful. I would find it helpful as a future reader. The goal of adding a comment is to capture the reason the method was originally created, so a reader or future maintainer can get into the original author's mindset at the time the code was written.

Reply:
Alright, makes sense. Just added the comment. Also realized using `metav1.Object` instead of `runtime.Object` may not make sense at the first glance, so, I added an explanation for that as well.

Reply:
Awesome. The `The reason metav1.Object is used . . .` is really helpful. For the other part, this part is a bit redundant, since it's in the method signature:

```go
// ConvertToUnstructured takes a Kubernetes object and converts it into
// *unstructured.Unstructured
```

I liked this part from your earlier explanation, so I think it'd make sense to have it in the comment:

> KubernetesApplication type requires you to send your objects as *unstructured.Unstructured instead of directly Deployment or Service

For example, something like:

```go
// ConvertToUnstructured exists because the KubernetesApplication type requires you 
// to send your objects as *unstructured.Unstructured instead of directly as Deployment or Service.
```

But, I don't think it's worth blocking the merge over; we can improve it in the future.